app.js

```diff
@@ -1 +1,52 @@
-console.log("js file is connected")
\ No newline at end of file
+checkedRadio = document.querySelector("input[name='ingredient']:checked");
+searchIngredient(checkedRadio);
+
+/**
+ * This function adds and ingredient a url, that is used to 
+ * make the call to the API for birthday cakes with that ingredient
+ * 
+ * @param {object} radioInput radio input DOM Element
+ */
+function searchIngredient(radioInput) {
+  const rootUrl = "https://api.edamam.com/search";
+  const recipeUrlElement = document.querySelector("#recipe-url");
+  const labelElement = document.querySelector("#label");
+  const imageElement = document.querySelector("#cake-img");
+  const ingredientsListElement = document.querySelector(".ingredients");
+
+  ingredientsListElement.innerHTML = "";
+  const apiUrl = `${rootUrl}?q=birthday+cake+${radioInput.value}&app_id=589ecbd6&app_key=6d6116bfcbdc60fe641222727dc9eb8f`;
+  fetch(apiUrl)
+    .then((response) => response.json())
+    .then((data) => {
+      console.log("Success:", data);
+      // The default number of search is 10 items, i.e when "from=" is specified but "to=" is not,
+      // or when neither "to=" or "from=" are specified in the apiUrl.
+      // With a free plan we are allowed to request max of 100 items.
+      // console.log(data.hits.length);
+
```

Comment:
You need these variables here, so I would move the declaration here. But actually, as you use them only once, there is no need in creating a variable at all:
```javascript
document.querySelector("#recipe-url").setAttribute("href", data.hits[0].recipe.url);
document.querySelector("#label").innerHTML = data.hits[0].recipe.label;;
document.querySelector("#cake-img").src = data.hits[0].recipe.image;
```
It makes the code look shorter and easier to understand.

Reply:
I guess, this is also about last comment